controllers/transfers.controller.js

```diff
@@ -32,11 +32,43 @@ const debit = async (account_number, amount) => {
   );
 };
 
+/**
+ * Debits an account by an amount
+ *
+ * @param {String} account_number the account number of the account to be debited
+ * @param {Number} amount the amount to be debited
+ */
+const accountCheck = async (account_number, amount) => {
+  const check = await Accounts.findOne({ account_number });
+  if (check.account_balance < amount) {
+    throw new BadRequestError("Insufficient Funds");
+  } else {
+    debit(account_number, amount);
+  }
+};
+
 const transfer = async (req, res) => {
   // debit the sender
   // credit the recipient
   // create the transaction
   // return a response ot throw and error
+  await accountCheck(req.body.sender, req.body.amount);
+  credit(req.body.recipient, req.body.amount);
+  const transaction = await Transaction.create(req.body);
+
+  res.status(StatusCodes.CREATED).json({ transaction });
+};
+
+const deposit = async (req, res) => {
+  const transaction = await Transaction.create(req.body);
+  credit(req.body.recipient, req.body.amount);
+  res.status(StatusCodes.CREATED).json({ transaction });
+};
+
+const withdrawal = async (req, res) => {
```

Comment:
Looks like you're creating a Transaction here AND in the `accountCheck` function. Is this a requirement?

Reply:
The accountCheck function checks if there is money in that account before debiting

Reply:
Yeah, but on line 65 you are calling `await Transaction.create(req.body)`